spark/datadog_checks/spark/spark.py

```diff
@@ -119,6 +119,8 @@ def check(self, _):
                 AgentCheck.OK,
                 tags=['url:%s' % am_address] + tags,
             )
+        else:
+            self.log.warning('No running apps found. No metrics will be collected.')
```

Comment:
How about moving this to right after here? https://github.com/DataDog/integrations-core/blob/df8a8d81463bf6d9775f6e54f4b22264d9e004b0/spark/datadog_checks/spark/spark.py#L92

And returning early from the function.

Reply:
Indeed, we could do that because all the functions are only iterating on the list. It makes me think that it would even be better to iterate on the apps inside the `check` function itself and make the `_spark_job_metrics`, `_spark_stage_metrics`, etc... process one app at a time

Reply:
We could also add a test with no running apps to make sure everything is working properly in this case